lsp-yaml.el

```diff
@@ -135,5 +153,68 @@
                                       (lsp--set-configuration
                                        (lsp-configuration-section "yaml"))))))
 
+(defconst lsp-yaml--built-in-kubernetes-schema
+  '((name . "Kubernetes")
+    (description . "Built-in kubernetes manifest schema definition")
+    (url . :kubernetes)
+    (fileMatch . ["*-k8s.yaml" "*-k8s.yml"])))
+
+(defun lsp-yaml-download-schema-store-db (&optional force-downloading)
+  "Download the remote schema store at `lsp-yaml-schema-store-uri' into local cache.
+Set FORCE-DOWNLOADING to non-nil to force re-download the database."
+  (interactive "P")
+  (or force-downloading
+      (file-exists-p lsp-yaml-schema-store-local-db)
+      (url-copy-file lsp-yaml-schema-store-uri lsp-yaml-schema-store-local-db)))
+
+(defun lsp-yaml--get-supported-schemas ()
+  "Get out the list of supported schemas."
+  (progn
+    (if lsp-yaml-schema-store-enable
+        (if (not lsp-yaml--schema-store-schemas-alist)
+            (progn
+              (lsp-yaml-download-schema-store-db)
+              (setq lsp-yaml--schema-store-schemas-alist
+                    (alist-get 'schemas (json-read-file lsp-yaml-schema-store-local-db))))))
+    (seq-concatenate 'list (list lsp-yaml--built-in-kubernetes-schema) lsp-yaml--schema-store-schemas-alist)))
+
+(defun lsp-yaml--helm-schema-candidates ()
+  "Build helm source for schema candidates."
+  (cl-loop for schema in (lsp-yaml--get-supported-schemas)
+	   collect (cons
+		    (alist-get 'description schema)
+		    schema)))
+
+(defun lsp-yaml-set-buffer-schema (uri)
+  "Set yaml schema for the current buffer to URI."
+  (let* ((workspace-path (file-relative-name
+                          (lsp--uri-to-path (lsp--buffer-uri))
+                          (lsp-workspace-root (lsp--buffer-uri))))
+         (glob-pattern (concat "/" workspace-path))
```

Comment:
I'm not quite sure about this since this is the recommended glob-pattern by yaml-language-server.
It would be parsed by the server and will be interpreted there as well.
https://github.com/redhat-developer/yaml-language-server

Reply:
I guess @kiennq  might test that on windows. Seems like `(lsp--buffer-uri)` might return uri with `\` as a separator so the path might end up like: `/foo\bar\baz.txt`. This may work too, but it has to be validated.

Reply:
Beautiful. If someone could test it on windows then that would be great.

Reply:
It's fine, `(lsp--buffer-uri)` will return `/` on Windows so no need to worry.